### PR TITLE
remove etcd2 client from server

### PR DESCRIPTION
**Pull request by deads2k**
We stopped supporting etcd2 in 3.7.  This remove the related code.

### CODE REVIEW

File: test/integration/etcd_storage_path_test.go [hunk 1]
```diff
@@ -963,41 +962,17 @@ var kindWhiteList = sets.NewString(
...
-/*
 func TestEtcd3StoragePath(t *testing.T) {
-	etcdServer, _ := testutil.RequireEtcd3(t)
-	defer testutil.DumpEtcdOnFailure(t)
-
-	getter := &etcd3Getter{
-		kv: etcdServer.V3Client,
-	}
-	testEtcdStoragePath(t, etcdServer, getter)
+	testEtcdStoragePath(t)
```
Comment (enj) [inline test/integration/etcd_storage_path_test.go]: Collapse these two functions.

File: test/integration/etcd_storage_path_test.go [hunk 2]
```diff
@@ -1050,6 +1022,12 @@ func testEtcdStoragePath(t *testing.T, etcdServer *etcdtest.EtcdTestServer, gett
 		t.Fatalf("error creating test namespace: %#v", err)
 	}
 
+	etcdClient3, err := etcd.MakeEtcdClientV3(masterConfig.EtcdClientInfo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	getter := &etcd3Getter{kv: etcdClient3.KV}
```
Comment (enj) [inline test/integration/etcd_storage_path_test.go]: The etcd3Getter struct seems a bit pointless now since you no longer have the interface.

**Comment by enj** [comment]
Minor nits.  I am confident that this did not break anything since the storage path test is still passing.

**Comment by deads2k** [comment]
comments addressed

**Pull request merged**